app/services/result.rb

```diff
@@ -96,6 +114,14 @@ def and_then(&block)
       self
     end
 
+    # Error a -> Result b
+    def rescue(&block)
+      result = block.call(self)
```

Comment:
The Result object is now passed to the handler block is as, which is different what `and_then` does. `and_then` passes unwrapped `data` to the handler. So I think `rescue` should also pass unwrapped `data` and the `error_msg` the same way as `on_error` does: https://github.com/sharetribe/sharetribe/pull/2559/files#diff-3806624eafa2c3ce0e7273410ae8e337R131

Does it make sense?